src/Application.php

```
@@ -245,6 +250,11 @@ public function handle(ServerRequestInterface $request, ResponseInterface $respo
             $handler  = $this->getInjector()->make($this->getActionHandler());
             $response = $handler($request, $response, $route);
 
+            foreach ($this->getMiddleware() as $middleware) {
+                $middleware = $this->getInjector()->make($middleware);
+                list($request, $response) = $middleware($request, $response);
```

This was an incomplete solution, but that's something we will have to figure out. Could we just throw an exception if we want to terminate out of the loop?